### ISSUE TITLE
weave 2.0.0 doesn't work when docker tls enabled

### ISSUE THREAD

**Issue by c4po** [weaveworks/weave#3041]
## What happened?

try to join weave to peer with command
```
/usr/local/bin/weave launch --ipalloc-range 100.32.0.0/12 10.115.169.4
```

```
INFO: 2017/06/26 15:38:43.389284 Command line options: map[dns-listen-address:172.17.0.1:53 host-root:/host ipalloc-range:100.32.0.0/12 proxy:true status-addr:127.0.0.1:6782 datapath:datapath plugin:true resolv-conf:/var/run/weave/etc/resolv.conf weave-bridge:weave docker-bridge:docker0 H:[unix:///var/run/weave/weave.sock] http-addr:127.0.0.1:6784 nickname:OCLD1LXONDEMAND40-2 port:6783 dns-effective-listen-address:172.17.0.1]
INFO: 2017/06/26 15:38:43.389372 weave  2.0.0
INFO: 2017/06/26 15:38:43.391088 Docker API on unix:///var/run/docker.sock: &[BuildTime=2016-04-27T00:34:42.305049348+00:00 Version=1.11.1 ApiVersion=1.23 GitCommit=5604cbe GoVersion=go1.5.4 Os=linux Arch=amd64 KernelVersion=3.10.0-514.6.1.el7.x86_64]
INFO: 2017/06/26 15:38:43.391284 Using docker bridge IP for DNS: 172.17.0.1
INFO: 2017/06/26 15:38:43.395256 proxy listening on unix:///var/run/weave/weave.sock
INFO: 2017/06/26 15:38:43.397173 Bridge type is bridged_fastdp
INFO: 2017/06/26 15:38:43.397196 Communication between peers is unencrypted.
INFO: 2017/06/26 15:38:43.411174 Our name is ca:0f:24:78:58:d4(OCLD1LXONDEMAND40-2)
INFO: 2017/06/26 15:38:43.411203 Launch detected - using supplied peer list: [10.115.169.4 --tlsverify --tlscacert /home/weave/tls/ca.pem --tlscert /home/weave/tls/cert.pem --tlskey /home/weave/tls/key.pem]
INFO: 2017/06/26 15:38:43.419520 Docker API on unix:///var/run/docker.sock: &[KernelVersion=3.10.0-514.6.1.el7.x86_64 BuildTime=2016-04-27T00:34:42.305049348+00:00 Version=1.11.1 ApiVersion=1.23 GitCommit=5604cbe GoVersion=go1.5.4 Os=linux Arch=amd64]
INFO: 2017/06/26 15:38:43.419919 Checking for pre-existing addresses on weave bridge
INFO: 2017/06/26 15:38:43.459692 [allocator ca:0f:24:78:58:d4] Initialising with persisted data
INFO: 2017/06/26 15:38:43.461335 Listening for DNS queries on 172.17.0.1
INFO: 2017/06/26 15:38:43.468257 Sniffing traffic on datapath (via ODP)
FATA: 2017/06/26 15:38:43.469333 lookup --tlsverify: no such host
lookup --tlscacert: no such host
lookup /home/weave/tls/ca.pem: no such host
lookup --tlscert: no such host
lookup /home/weave/tls/cert.pem: no such host
lookup --tlskey: no such host
lookup /home/weave/tls/key.pem: no such host
```

## How to reproduce it?

in my docker.service file
docker is start with these parameters
```
/usr/bin/docker daemon -H fd:// -H unix:///var/run/docker.sock -H tcp://0.0.0.0:2376
              --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server-cert.pem --tlskey /etc/docker/server-key.pem
```

## Versions:

```
$ weave version
weave script 2.0.0
weave 2.0.0

$ docker version
Client:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:42 2016
 OS/Arch:      linux/amd64
$ uname -a
Linux OCLD1LXONDEMAND40-2 3.10.0-514.6.1.el7.x86_64 #1 SMP Wed Jan 18 13:06:36 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by marccarre** [comment]
Hi @c4po, many thanks for raising this issue! :+1: 
It looks like a duplicate of #3034, for which we're already on a fix, so I'll close this one for now.

**Comment by marccarre** [comment]
Fixed by #3039
